src/corrgi/alignment.py

```diff
@@ -0,0 +1,62 @@
+import itertools
+
+import pandas as pd
+from hipscat.catalog import Catalog
+from hipscat.pixel_tree.pixel_alignment import PixelAlignment
+from hipscat.pixel_tree.pixel_alignment_types import PixelAlignmentType
+
+column_names = [
+    PixelAlignment.PRIMARY_ORDER_COLUMN_NAME,
+    PixelAlignment.PRIMARY_PIXEL_COLUMN_NAME,
+    PixelAlignment.JOIN_ORDER_COLUMN_NAME,
+    PixelAlignment.JOIN_PIXEL_COLUMN_NAME,
+    PixelAlignment.ALIGNED_ORDER_COLUMN_NAME,
+    PixelAlignment.ALIGNED_PIXEL_COLUMN_NAME,
+]
+
+
+def autocorrelation_alignment(catalog: Catalog) -> PixelAlignment:
+    """Determine all pairs of partitions that should be correlated within the same catalog.
+
+    This considers all combinations, without duplicates between the "primary" and "join"
+    pixels in the alignment.
+
+    Args:
+        catalog (Catalog): catalog for autocorrelation
+    Returns:
+        alignment object where the `aligned` columns simply match the left pixel.
+    """
+    upper_triangle = [
+        [left.order, left.pixel, right.order, right.pixel, left.order, left.pixel]
+        for (left, right) in itertools.combinations(catalog.get_healpix_pixels(), 2)
+    ]
+    upper_triangle = pd.DataFrame(upper_triangle, columns=column_names)
+    diagonal = pd.DataFrame(
+        [
+            [pix.order, pix.pixel, pix.order, pix.pixel, pix.order, pix.pixel]
+            for pix in catalog.get_healpix_pixels()
+        ]
```

Comment:
We need to assign the column names here too:
```suggestion
        ], columns=column_names
```